ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgentTest.java

```diff
@@ -898,6 +905,27 @@ public void assertHostCorrectlyRevertsOnIgnoredPacketFrom() {
     assertThat(PeerDiscoveryAgent.deriveHost(source, mockWellFormed)).isEqualTo(routableHost);
   }
 
+  @Test
+  void testFromEnodeWithDiscoveryDisabled()  {
+      EnodeURL enodeWithNoDiscovery = mock(EnodeURL.class);
+      when(enodeWithNoDiscovery.getDiscoveryPort()).thenReturn(Optional.empty());
+      when(enodeWithNoDiscovery.getListeningPort()).thenReturn(Optional.of(8545));
+      try {
+        when(enodeWithNoDiscovery.getIp()).thenReturn(InetAddress.getByName("127.0.0.1"));
+      }
+      catch (UnknownHostException e) {
+        log.debug("Failed to resolve the Host Address ");
```

Comment:
prob wouldn't catch this exception in the test, you can mark the test method throws that ex. (what would happen if this exception occurred? would the test pass or fail? should it pass or fail?)

Reply:
I don't believe this will significantly impact the test's primary function. Should I proceed with Throws it?

Reply:
yes I would keep this logic out of the test altogether